src/ordinal.rs

```diff
@@ -107,9 +107,17 @@ impl Ordinal {
 
     let cycle_start_epoch = cycle_number * CYCLE_EPOCHS;
 
-    let cycle_progression = period_offset
-      .checked_sub(epoch_offset % PERIOD_BLOCKS)
-      .ok_or_else(|| anyhow!("Invalid relationship between epoch offset and period offset"))?;
+    // This checked sub is returning None so `ok_or_else` returns
+    // an error:
+    //
+    // For some reason I thought this would never be negative, why?
+    //
+    // epoch_offset % PERIOD_BLOCKS can be 2015
+    //
+    // - What is the earliest ordinal that has this problem?
+    // - Why did I assume that this checked_sub would always succeed?
+    let cycle_progression =
+      (((period_offset as i64) - ((epoch_offset % PERIOD_BLOCKS) as i64)).abs()) as u64;
 
     if cycle_progression % (Epoch::BLOCKS % PERIOD_BLOCKS) != 0 {
       bail!("Invalid relationship between epoch offset and period offset");
```

Comment:
Not sure what `cycle_progression` signifies exactly. (number of block in current cycle?)